### ISSUE TITLE
Panic Error when hashicorp is not available

### ISSUE THREAD

**Issue by cbuschka** [cbuschka/tfvm#44]
* [x] This is a bug report
* [ ] This is a feature request

### Expected behavior

Install terraform.

### Actual behavior

It segfaulted.

### Steps to reproduce the behavior

Currently not possible

**Output of `tfvm info`:**

```
Terraform version is 1.3.6.
Initializing terraform...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x669eed]

goroutine 1 [running]:
github.com/cbuschka/tfvm/internal/remote.getHTMLContentsFrom({0xc000183200, 0x33})
	/home/conni/work/fcd/github/cbuschka/tfvm/internal/remote/hashicorp.go:60 +0x32d
github.com/cbuschka/tfvm/internal/remote.downloadReleasesPage()
	/home/conni/work/fcd/github/cbuschka/tfvm/internal/remote/hashicorp.go:132 +0xaa
github.com/cbuschka/tfvm/internal/remote.(*hashicorpRemote).ListTerraformReleases(0x7017ed?)
	/home/conni/work/fcd/github/cbuschka/tfvm/internal/remote/hashicorp.go:30 +0x19
github.com/cbuschka/tfvm/internal/inventory.(*Inventory).Update(0xc0000a62a0)
	/home/conni/work/fcd/github/cbuschka/tfvm/internal/inventory/updating.go:13 +0x49
github.com/cbuschka/tfvm/internal/commands.RunTerraformCommand({0xc00008e160, 0x2, 0x2})
	/home/conni/work/fcd/github/cbuschka/tfvm/internal/commands/tfvm_terraform_command.go:41 +0x149
github.com/cbuschka/tfvm/internal/cli.Run({0xc00008e[15](https://github.com/otto-ec/bu_dropstock_env/actions/runs/5542539704/jobs/10117353853#step:8:16)0?, 0x3, 0x3})
	/home/conni/work/fcd/github/cbuschka/tfvm/internal/cli/run.go:[22](https://github.com/otto-ec/bu_dropstock_env/actions/runs/5542539704/jobs/10117353853#step:8:23) +0x358
main.main()
	/home/conni/work/fcd/github/cbuschka/tfvm/cmd/tfvm/tfvm.go:11 +0x2e
```

**Additional environment details**

Github actions env.

Original issue posted as https://github.com/cbuschka/setup-tfvm/issues/13 by @LeonStoldt

**Comment by cbuschka** [comment]
`        resp, err := client.Do(req)
        if resp.StatusCode != 200 {
                log.Debugf("Response to GET for '%s': %d/%s", url, resp.StatusCode, resp.Status)
                return "", fmt.Errorf("downloading %s failed", url)
        }

        if err != nil {
                return "", err
        }
`

Looks like resp is used before err is checked. Will fix it. Thanks for reporting back <3

**Comment by cbuschka** [comment]
Ok, fixed. Will be available by v0.25.0.
